packages/react/src/providers/Provider.tsx

```diff
@@ -61,22 +65,32 @@ export function Provider ({
     }
   }, [client, events])
 
-  const getClient = async (): Promise<Client> => {
-    if (client == null) {
-      const { client, events } = await createClient({ servicePrincipal, connection })
-      setClient(client)
-      setEvents(events)
-      setAccounts(Object.values(client.accounts()))
-      setSpaces(client.spaces())
-      return client
-    }
-    return client
+  const setupClient = async (): Promise<void> => {
+    const { client, events } = await createClient({ servicePrincipal, connection })
+    setClient(client)
+    setEvents(events)
+    setAccounts(Object.values(client.accounts()))
+    setSpaces(client.spaces())
+  }
+
+  const logout = async (): Promise<void> => {
+    // it's possible that setupClient hasn't been run yet - run createClient here
+    // to get a reliable handle on the latest store
+    const { store } = await createClient({ servicePrincipal, connection })
+    await store.reset()
+    // set state back to defaults
+    setClient(undefined)
+    setEvents(undefined)
+    setAccounts([])
+    setSpaces([])
+    // set state up again
+    await setupClient()
   }
 
-  useEffect(() => { void getClient() }, []) // load client - once.
+  useEffect(() => { void setupClient() }, []) // load client - once.
```

Comment:
if props.connection changes, wouldn't we want this to run again? so that a new client gets created and passed to `setClient`?

maybe `setupClient` assignment should use `React.createMemo` with appropriate dependencies, and this `useEffect` can depend on the variables it depends on e.g. `setupClient`

Reply:
also maybe we should have the lint rule for react hook deps?  https://github.com/facebook/react/blob/main/packages/eslint-plugin-react-hooks/README.md

Reply:
ah, I think you're right! I think that's orthogonal to the purpose of this PR though - happy to file an issue and even work on it today, but want to get this PR merged and shipped to fix the regression in production